pkg/pma/pma.c

```diff
@@ -0,0 +1,783 @@
+#include "pma.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "page.h"
+#include "sigsegv.h"
+#include "util.h"
+#include "wal.h"
+
+#ifdef PMA_TEST
```

Comment:
I haven't seen this pattern before, is it normal in "regular" C programs?

Reply:
I'd presume this was done so that a test compiled as another compilation unit could access normally static (no longer) identifiers. It doesn't actually seem to be used though, at least currently. And I think a better pattern would be to avoid this and to just have the test import `pma_tests.c` directly to access its static definitions if necessary.

Reply:
@barter-simsum that's largely correct. `PMA_TEST` was defined via the `test` configuration (i.e. `--config=test`), but you make a good point about directly including `pma.c` for access to `static` functions. I've changed `pma_tests.c` to directly include `pma.c`.